archiver/biosamples.py

```diff
@@ -0,0 +1,42 @@
+from archiver.dsp_post_process import dsp_attribute, fixed_dsp_attribute, taxon_id
+from conversion.json_mapper import JsonMapper
+from conversion.post_process import format_date, default_to
+
+
+def _taxon(*args):
+    ontology_item = args[0]
+    if ontology_item:
+        genus_species = ontology_item[0]
+        return genus_species.get('ontology_label')
+
+
+def derive_concrete_type(*args):
+    schema_url = args[0]
+    concrete_type = schema_url.split('/')[-1]
+    return dsp_attribute(concrete_type)
+
+
+spec = {
+    '$on': 'biomaterial',
+    'alias': ['uuid.uuid'],
+    'attributes': {
+        'Biomaterial Core - Biomaterial Id': ['content.biomaterial_core.biomaterial_id', dsp_attribute],
+        'HCA Biomaterial Type': ['content.describedBy', derive_concrete_type],
+        'HCA Biomaterial UUID': ['uuid.uuid', dsp_attribute],
+        'Is Living': ['content.is_living', dsp_attribute],
```

Comment:
Guessing the integ. test error is caused by this mapping being applied to all biomaterials, so entities like cell_suspension won't have an is_living field. Should we have mapping files per concrete HCA entity, rather than for Biomaterials as a whole?

A quick fix would be to potentially omit the is_living field if it's not present in the source metadata (I imagine the DSP validator doesn't require the is_living attribute, but when it is present it's value should be non-empty). Could augment the json-mapper to accept an "omit_if_empty" function to cancel the mapping attempt for the empty 'content.is_living'.

I think the safest bet in the long run is to have mappings targeted at concrete HCA entities